Amend new-hub infrastructure docs

yuvipanda opened the pull request:
Ref #2062

yuvipanda committed:
Amend new-hub infrastructure

Ref #2062

yuvipanda committed:
Default to datahub image for new hubs

yuvipanda committed:
Include encrypted bcourses secrets in template

yuvipanda committed:
Document adding deploy step to circleci

felder commented:
@yuvipanda Looking over the docs:

1. jupyterhub.hub.cookieSecret is missing entirely.

2. Also it's unclear from the documentation whether or not the same values are placed in the secrets file for both production and staging or if they require different ones.  The docs appear to indicate prod and staging should be done as two separate procedures, but then the docs ALSO tell you to fill in values for staging and prod at the same time (step 5 and step 13). I think the docs would be clearer if they indicated which tasks have to be performed once (I assume cookiecutter is once) and which ones need to be performed separately for staging and prod.

3. Step 7 is missing the actual example stanza.

4. Link in step 9 is broken.

felder commented:
@yuvipanda 

I think I'm going to surrender on attempting to create dlab otherwise I'm going to have to bug Ryan throughout the afternoon.

IMO, the current docs are not sufficient to allow someone who has not done this before to succeed.

yuvipanda committed:
Automate secret key generation for templates

cookiecutter now has 'hooks' that can run after they
generate the files. We use this to generate secret keys
and remove many manual steps.

The docs have also been updated to be more
accurate.

yuvipanda committed:
Move the default auth to canvas

The default is now canvas, not google auth. We should
migrate biology too at some point.

yuvipanda committed:
Remove all unused google auth credentials

Was causing a lot of confusion!

yuvipanda committed:
Self-document the secrets template YAML

yuvipanda committed:
Cleanup text in new hub documentation

yuvipanda commented:
Thank you for all this feedback, @felder. Sorry it was a frustrating experience :( 

I've updated this PR with new docs based on your feedback here and on slack. I've also automated *all* secret generation, so the hub setup process is much simpler!

yuvipanda committed:
Get traffic into cs194 staging hub with ingresses

- Remove the explicit DNS entry for cs194-staging,
  so the wildcard entry resolves it to our nginx-ingress.
- Setup ingress config for cs194-staging, so HTTPS is autogenerated.
- Manually allocate SSH / SFTP ports for cs194-staging.
  nginx-ingress will proxy incoming ssh connections to the correct
  hub based on the ports folks are connecting via.
- jupyterhub-ssh helm chart bump so we can allow traffic from
  nginx-ingress via networkpolicy

I'll deploy manually to a few hubs, and slowly roll it out
across all the hubs.

Ref #2167

yuvipanda committed:
Remove instructions on DNS & static IP sets

We have a wildcard domain from *.datahub.berkele.edu
to our nginx-ingress IP. This simplifies new hub creation a
*lot*.

Ref #2167

yuvipanda committed:
Mention possible delay for https setup
